python-api/app/api/v1/openapi/router.py

```
@@ -1,66 +1,68 @@
 @router.get("/generations", response_model=GenerationResponse, tags=["OpenRouter"])
 def get_generation(generation_id: str = Query(..., description="ID of the generation")):
     try:
         return client.get_generation(generation_id)
-    except requests.HTTPError as http_err:
-        if http_err.response.status_code == 404:
-            raise HTTPException(status_code=404, detail=f"Generation ID '{generation_id}' not found.")
+    except LLMError as llm_err:
+        if llm_err.status_code == 404 or "404" in str(llm_err):
+            raise HTTPException(
+                status_code=404, detail=f"Generation ID '{generation_id}' not found."
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)